### PR TITLE
Fix Lottery API

### PR DESCRIPTION
**Pull request by CrVnsh**
This PR will fix the Problem that the API is crashing whenever more than 125 Lotteries are getting requested at the same time.
The prevent it from crashing the pr is including 2 fixes:
- a connected check implemented with the Singleton pattern
- web3 feature batchrequest
  - this will allow the api to make multiple requests with just one call against the bsc api

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
     "@types/node": "^14.14.7",
-    "@now/node": "^1.3.0",
+    "@vercel/node": "^1.8.5",
     "jest": "^26.6.3",
     "ts-jest": "^26.4.4",
     "ts-node": "^9.0.0"
   },
   "scripts": {
-    "dev": "gatsby develop",
-    "build": "gatsby build",
```
Comment (pancake-swap) [inline package.json:26]: so what's the command for build then?

Comment (CrVnsh) [reply package.json:26]: You don't need a build command vercel is using the default nodejs api build command.
To build it on your local machine you can run vercel dev

File: api/pending.ts [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
-}
+  const displayBalance = balance.dividedBy(new BigNumber(10).pow(decimals));
+  return displayBalance.toNumber();
+};
 
 const web3 = new Web3(
-    new Web3.providers.HttpProvider(
-        "https://bsc-dataseed.binance.org"
-    )
+  new Web3.providers.HttpProvider("https://bsc-dataseed.binance.org")
```
Comment (ChefKai) [inline api/pending.ts:13]: Is it possible to use the Web3 Provider that is exported from the `lib`?

**Review by ChefKai — COMMENTED** [review]
LGTM

Thanks for the `@now` -> `@vercel` change as well!

File: api/pending.ts (hunk 2 above)
Comment (CrVnsh) [reply api/pending.ts]: Fixed with https://github.com/pancakeswap/gatsby-pancake-api/pull/3/commits/6fd8b808b5277a07867d869df54ddec098c8e616

File: lib/eth.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Web3 from "web3";
+import { getProvider } from "./provider";
+import { HttpProvider as Web3HttpProvider } from "web3-providers-http";
```
[the remark this answers is not in the document]
Comment (CrVnsh) [reply lib/eth.ts]: fixed with https://github.com/pancakeswap/gatsby-pancake-api/pull/3/commits/6fd8b808b5277a07867d869df54ddec098c8e616

**Pull request merged**